staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -374,184 +370,26 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 }
 
 func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
-	if len(o.Selector) == 0 {
-		return nil, fmt.Errorf("selector must be specified")
+	opts := versioned.Options{
+		Name:            o.Name,
+		DefaultName:     o.DefaultName,
+		Selector:        o.Selector,
+		Labels:          o.Labels,
+		Protocol:        o.Protocol,
+		Protocols:       o.Protocols,
+		Port:            o.Port,
+		Ports:           o.Ports,
+		PortName:        o.PortName,
+		TargetPort:      o.TargetPort,
+		ExternalIP:      o.ExternalIP,
+		Type:            o.Type,
+		LoadBalancerIP:  o.LoadBalancerIP,
+		SessionAffinity: o.SessionAffinity,
+		ClusterIP:       o.ClusterIP,
 	}
-	selector, err := parseLabels(o.Selector)
+	obj, err := versioned.GenerateService(opts)
```

Comment:
This looks unusual. What is the reason of this change?

Reply:
Repetitive code implementation, which I extract as a public method

Reply:
See my comment above.  We don't want to use the generators.